### PR TITLE
Add several examples of trainingjob

### PR DESCRIPTION
**Pull request by xyxxxxx**

### CODE REVIEW

File: job/mpijob/horovod-keras/job.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    template:
+      spec:
+        containers:
+          - name: mpi-worker
+            image: registry.tensorstack.cn/t9k/horovod:20211201
+            imagePullPolicy: IfNotPresent
+            workingDir: /t9k/mnt/tutorial-examples
+            env:
+              - name: LD_LIBRARY_PATH
+                value: "/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/lib"
```
Comment (ppkube) [inline job/mpijob/horovod-keras/job.yaml:35]: Does this example use GPU? If yes, there should be extended resource requests, if not, these cuda/nvidia related paths are useless. 

env `NCCL_DEBUG` is for nvidia lib too, is it really required?

File: job/mpijob/horovod-torch/job.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      - "--no_cuda"
+    template:
+      spec:
+        containers:
+          - name: mpi-worker
+            image: registry.tensorstack.cn/t9k/horovod:20211201
+            imagePullPolicy: IfNotPresent
+            workingDir: /t9k/mnt/tutorial-examples
+            env:
+              - name: LD_LIBRARY_PATH
```
Comment (ppkube) [inline job/mpijob/horovod-torch/job.yaml:34]: ditto

File: job/pytorchtrainingjob/ddp/job.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: PyTorchTrainingJob
+metadata:
+  name: torch-mnist-trainingjob
+spec:
+  tensorboardSpec:
+    trainingLogFilesets:
+      - t9k://pvc/tutorial/tutorial-examples/job/pytorchtrainingjob/ddp/log
+    image: registry.tensorstack.cn/t9k/tensorflow-2.5.0:cpu
```
Comment (ppkube) [inline job/pytorchtrainingjob/ddp/job.yaml:9]: 2.5 is pretty old. can we migrate to a newer release of tensorflow?

File: job/pytorchtrainingjob/ps/job.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
+apiVersion: batch.tensorstack.dev/v1beta1
+kind: PyTorchTrainingJob
+metadata:
+  name: torch-mnist-trainingjob-ps
+spec:
+  scheduler:
```
Comment (ppkube) [inline job/pytorchtrainingjob/ps/job.yaml:6]: let's make this set of samples simple, and refrain from using `spec.scheduler` for now

File: job/tensorflowtrainingjob/ps/job.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            runAsUser: 1000
+          containers:
+            - command:
+                - python3
+                - job/tensorflowtrainingjob/ps/keras_mnist_trainingjob_ps.py
+                - "--log_dir"
+                - "job/tensorflowtrainingjob/ps/log"
+                - "--no_cuda"
+              workingDir: /t9k/mnt/tutorial-examples
+              image: registry.tensorstack.cn/t9k/tensorflow-2.5.0:cpu
```
Comment (ppkube) [inline job/tensorflowtrainingjob/ps/job.yaml:26]: use newer release of tensorflow?

File: job/tensorflowtrainingjob/multiworker/README.md [hunk 6]
```diff
@@ -2,17 +2,11 @@
 
 本示例使用 TensorFlowTrainingJob 对 Keras 模型进行多工作器同步训练（采用 [`tf.distribute.MultiWorkerMirroredStrategy`](https://www.tensorflow.org/api_docs/python/tf/distribute/MultiWorkerMirroredStrategy) 分布式策略）。
 
-切换到当前目录下，执行 `download_dataset.py` 脚本以下载数据集：
+切换到当前目录下，使用 `job.yaml` 创建 TensorFlowTrainingJob：
 
 ```shell
 # cd into current directory
 cd ~/tutorial-examples/job/tensorflowtrainingjob/multiworker
-python download_dataset.py
```
Comment (ppkube) [inline job/tensorflowtrainingjob/multiworker/README.md:10]: why removing this download step?

**Review by ppkube — CHANGES_REQUESTED** [review]

File: job/tensorflowtrainingjob/multiworker/README.md (hunk 6 above)
Comment (xyxxxxx) [reply job/tensorflowtrainingjob/multiworker/README.md:10]: 因为我意识到这一步需要 notebook 安装 `tensorflow` 包，对于 pytorch 的示例，这一步需要安装 `torchvision` 包，对于其他的示例，可能还需要安装别的包

由于没有一个 notebook 镜像有所有这些包，所以我把下载数据集的步骤放到脚本里，就不再限制 notebook 使用哪个镜像

Comment (ppkube) [reply job/tensorflowtrainingjob/multiworker/README.md:10]: 放到那个 script 里了？

File: job/mpijob/horovod-keras/job.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      - python
+      - job/mpijob/horovod-keras/keras_mnist_mpijob.py
+      - "--log_dir"
+      - "job/mpijob/horovod-keras/log"
+      - "--no_cuda"
+    template:
+      spec:
+        containers:
+          - name: mpi-worker
+            image: registry.tensorstack.cn/t9k/horovod:20211201
```
Comment (ppkube) [inline job/mpijob/horovod-keras/job.yaml:30]: is this image available? I got 'NOT Found'

File: job/pytorchtrainingjob/ddp/job.yaml (hunk 3 above)
Comment (xyxxxxx) [reply job/pytorchtrainingjob/ddp/job.yaml:9]: 一样跑没有区别，但是 `registry.tensorstack.cn` 没有新版本的镜像，我弄一个 2.11.0 的然后 push 上去

File: job/mpijob/horovod-keras/job.yaml (hunk 7 above)
Comment (xyxxxxx) [reply job/mpijob/horovod-keras/job.yaml:30]: 这个镜像没有 push 到 `registry.tensorstack.cn` 上，我也搞个新的镜像再 push 上去

测试的话可以用 `tsz.io/t9k/horovod:20211201`

File: job/mpijob/horovod-keras/job.yaml (hunk 1 above)
Comment (xyxxxxx) [reply job/mpijob/horovod-keras/job.yaml]: removed

File: job/mpijob/horovod-torch/job.yaml (hunk 2 above)
Comment (xyxxxxx) [reply job/mpijob/horovod-torch/job.yaml]: removed

File: job/pytorchtrainingjob/ddp/job.yaml (hunk 3 above)
Comment (xyxxxxx) [reply job/pytorchtrainingjob/ddp/job.yaml]: change to `registry.tensorstack.cn/t9k/tensorflow-2.11.0:cpu-sdk-0.5.1`

File: job/tensorflowtrainingjob/ps/job.yaml (hunk 5 above)
Comment (xyxxxxx) [reply job/tensorflowtrainingjob/ps/job.yaml]: change to `registry.tensorstack.cn/t9k/tensorflow-2.11.0:cpu-sdk-0.5.1`

File: job/mpijob/horovod-keras/job.yaml (hunk 7 above)
Comment (xyxxxxx) [reply job/mpijob/horovod-keras/job.yaml]: change to `registry.tensorstack.cn/t9k/horovod:sdk-0.5.1`

File: job/tensorflowtrainingjob/multiworker/README.md (hunk 6 above)
Comment (xyxxxxx) [reply job/tensorflowtrainingjob/multiworker/README.md:10]: https://github.com/t9k/tutorial-examples/blob/xyx/1223/job/tensorflowtrainingjob/multiworker/keras_mnist_trainingjob.py#L50

数据集不存在时自动下载

File: job/mpijob/horovod-keras/job.yaml (hunk 1 above)
Comment (ppkube) [reply job/mpijob/horovod-keras/job.yaml]: so it's been verified that the sample works after removing these 2 env entries?

**Review by ppkube — APPROVED** [review]
